File: web/plugins/range.js
import { WebPlugin } from "../engine/plugin.js";

export class RangePlugin extends WebPlugin {
    constructor() {
        super("範囲");
    }

    async execute(line, context) {
        const expression = line.replace(/^範囲\s*/, "");
        const values = expression.split(",").map(v =>
            Number(context.parser.parseValue(v.trim(), context.runtime))
        );

        context.runtime.write(
            Math.max(...values) - Math.min(...values)
        );
    }
}

File: pc/plugins/range.py
import re
from core.plugin import Plugin


class RangePlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'範囲\s+(.+)', line)
        if not match:
            return None

        values = match.group(1)
        return f"print(max([{values}]) - min([{values}]))"